plugins/plugins.go

```diff
@@ -982,3 +1052,37 @@ func (m *Manager) RegisterNDCacheTrigger(trigger func(bool)) {
 	defer m.mtx.Unlock()
 	m.registeredNDCacheTriggers = append(m.registeredNDCacheTriggers, trigger)
 }
+
+func (m *Manager) sendOPAUpdateLoop(ctx context.Context, uploadDuration time.Duration) {
+	ticker := time.NewTicker(uploadDuration)
+	mr.New(mr.NewSource(time.Now().UnixNano()))
+
+	ctx, cancel := context.WithCancel(ctx)
+
+	var opaReportNotify bool
+
+	for {
+		select {
+		case <-m.opaReportNotifyCh:
+			opaReportNotify = true
+		case <-ticker.C:
+			ticker.Stop()
+
+			if opaReportNotify {
+				opaReportNotify = false
+				_, err := m.reporter.SendReport(ctx)
+				if err != nil {
+					m.logger.WithFields(map[string]interface{}{"err": err}).Debug("Unable to send OPA telemetry report.")
```

Comment:
[nit]
```suggestion
					m.logger.WithError(err).Debug("Unable to send OPA telemetry report.")
```

This yields the same log line **if** we also set [the logrus *global variable* ](https://github.com/sirupsen/logrus/blob/master/entry.go#L38)`logrus.ErrorKey = "err"` _somewhere_ in the code.